src/lib/libcore.js

```diff
@@ -1724,8 +1724,15 @@ addToLibrary({
   $dynCallLegacy__deps: ['$dynCalls'],
   $dynCallLegacy: (sig, ptr, args) => {
     sig = sig.replace(/p/g, {{{ MEMORY64 ? "'j'" : "'i'" }}})
+
+#if !DECLARE_ASM_MODULE_EXPORTS
+    var dynCallFn = globalThis[`dynCall_${sig}`];
+#else
+    var dynCallFn = dynCalls[sig];
```

Comment:
Doesn't the dynCalls export get populated, even in `DECLARE_ASM_MODULE_EXPORTS=0` mode?   

I can't see why it shouldn't be.

It looks simple enough to modify `exportWasmSymbols` to handle there `dynCall_` prefix (would only need to be `#if DYNCALLS || !WASM_BIGINT`).

Reply:
Very nice.. This didn't even occur to me, I was in the mindset that "of course the dyncalls would be statically emitted in DECLARE_ASM_MODULE_EXPORTS=1 and assigned from there to dynCalls, so that can't work."

Changed the implementation from `exportWasmSymbols()` to route the dynCalls over. That works much better, and also then resolves the dynamic linking + wasm_bigint case, for a simpler overall implementation.